doc/sources/array_api.rst

```diff
@@ -0,0 +1,120 @@
+`sklearnex` device offloading.
+
+.. note::
+    Current support for DPNP and DPCTL usm_ndarray data can be copied and moved to and from device in sklearnex and have
+    impacts on memory utilization.
+
+DPCTL or DPNP inputs doesn't require to use `config_context(target_offload=device)`.
+`sklearnex` will use input usm_ndarray sycl context for device offloading.
+
+.. note::
+    As DPCTL or DPNP inputs contain SYCL contexts, they do not require `config_context(target_offload=device)`.
+    However, the use of `config_context`` will override the contained SYCL context and will force movement
+    of data to the targeted device.
+
+
+Example usage
+=============
+
+DPNP ndarrays
+-------------
+
+Here is an example code to demonstrate how to use `dpnp <https://github.com/IntelPython/dpnp>`__ arrays to
+run `RandomForestRegressor` on a GPU witout ```config_context(array_api_dispatch=True)```:
+
+.. literalinclude:: ../../examples/sklearnex/random_forest_regressor_dpnp.py
+	   :language: python
+
+
+.. note::
+    Functional support doesn't guarantee that after the model is trained, fitted attributes that are arrays
+    will also be from the same namespace as the training data.
+
+For example, if `dpnp's <https://github.com/IntelPython/dpnp>`__ namespace was used for training,
+then fitted attributes will be on the CPU and `numpy.ndarray` data format.
+
+DPCTL usm_ndarrays
+------------------
+Here is an example code to demonstrate how to use `dpctl <https://intelpython.github.io/dpctl/latest/index.html>`__
+arrays to run `RandomForestClassifier` on a GPU witout `config_context(array_api_dispatch=True)`:
+
+.. literalinclude:: ../../examples/sklearnex/random_forest_classifier_dpctl.py
+	   :language: python
+
+As on previous example, if `dpctl <https://intelpython.github.io/dpctl/latest/index.html>`__ Array API namespace was
+used for training, then fitted attributes will be on the CPU and `numpy.ndarray` data format.
+
+Use of `array-api-strict`
+-------------------------
+
+Here is an example code to demonstrate how to use `array-api-strict <https://github.com/data-apis/array-api-strict>`__
+arrays to run `DBSCAN`.
+
+.. literalinclude:: ../../examples/sklearnex/dbscan_array_api.py
+	   :language: python
+
+
+Support for Array API-compatible inputs
+=======================================
+All patched estimators, metrics, tools and non-scikit-learn estimators functionally support Array API.
+Functionally all input and output data have the same data format type.
+For all array inputs except SYCL USM arrays `dpnp's <https://github.com/IntelPython/dpnp>`__ ndarray and
+`Data Parallel Control usm_ndarray <https://intelpython.github.io/dpctl/latest/index.html>`__ all computation
+will be only accomplished on CPU.
+
+Stock scikit-learn uses `config_context(array_api_dispatch=True)` for enabling Array API `support <https://scikit-learn.org/1.5/modules/array_api.html>`__.
```

Comment:
```suggestion
Scikit-learn uses `config_context(array_api_dispatch=True)` for enabling Array API `support <https://scikit-learn.org/1.5/modules/array_api.html>`__.
```

Reply:
Double check capitalization of Scikit-learn vs Scikit-Learn in this file, it could be that my suggestions to 'Scikit-Learn' are incorrect, but they need to be consistent.

Reply:
I would like to leave stock prefix here just for the context.
Everywhere in our docs and repository `Scikit-learn` is used. If updates are required then that alignment could be done as a follow up in separate PR

Reply:
What does stock mean in this case?

Reply:
Stock means base or original scikit-learn: https://scikit-learn.org/stable/

I am referencing other to other parts of our docs where this is used.

https://github.com/intel/scikit-learn-intelex/blob/b077b6c35c5fb8ff191c9ea4c636cf73964005ba/README.md?plain=1#L91

https://github.com/intel/scikit-learn-intelex/blob/b077b6c35c5fb8ff191c9ea4c636cf73964005ba/README.md?plain=1#L94

https://github.com/intel/scikit-learn-intelex/blob/b077b6c35c5fb8ff191c9ea4c636cf73964005ba/examples/sklearnex/verbose_mode.py#L18

https://github.com/intel/scikit-learn-intelex/blob/b077b6c35c5fb8ff191c9ea4c636cf73964005ba/doc/sources/tutorials.rst#L63

https://github.com/intel/scikit-learn-intelex/blob/b077b6c35c5fb8ff191c9ea4c636cf73964005ba/doc/sources/oneapi-gpu.rst#L64

https://github.com/intel/scikit-learn-intelex/blob/b077b6c35c5fb8ff191c9ea4c636cf73964005ba/doc/sources/kaggle/regression.rst#L21

https://github.com/intel/scikit-learn-intelex/blob/b077b6c35c5fb8ff191c9ea4c636cf73964005ba/doc/sources/quick-start.rst#L166

https://github.com/intel/scikit-learn-intelex/blob/b077b6c35c5fb8ff191c9ea4c636cf73964005ba/doc/sources/kaggle/classification.rst#L21

Also it is used in docstrings everywhere through the repo.